client/blocks/reader-excerpt/index.jsx

```diff
@@ -115,16 +116,21 @@ const chooseExcerpt = ( post ) => {
 	return null;
 };
 
-const ReaderExcerpt = ( { post } ) => {
+const ReaderExcerpt = ( { post, setHasExcerpt } ) => {
 	const isDailyPrompt = !! getDailyPromptText( post );
+	const excerpt = chooseExcerpt( post );
+
+	useEffect( () => {
+		setHasExcerpt( !! excerpt );
```

Comment:
Might it be clearer to check for some expected empty values?
```suggestion
		setHasExcerpt( excerpt !== '' && excerpt !== null && excerpt !== undefined ); // not sure which of these values are actually possible/expected
```

Reply:
Types would be nice here, huh? 😅 I think `''` and `null` are valid checks.